### PR TITLE
Switch to syscall abi

### PR DESCRIPTION
**Pull request by caden-kline**
syscall abi.

### CODE REVIEW

File: hypercall.h [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
-    register unsigned long a1 asm("a1") = arg1;
+    register unsigned long v0 asm("v0") = num;
+    register unsigned long a0 asm("a0") = arg1;
 
     asm volatile(
        "movz $0, $0, $0"
-        : "+r"(a0)  // Input and output in R0
-        : "r"(a1) // arg1 in register A1
+        : "+r"(v0)  // Input and output in R0
+        : "r"(a0) // arg1 in register A1
```
Comment (be32826) [inline hypercall.h:9]: Fix (or delete) these comments

File: hypercall.h [hunk 2]
```diff
@@ -66,17 +66,17 @@ static inline unsigned long igloo_hypercall2(unsigned long num, unsigned long ar
...
+    register unsigned long v0 asm("v0") = num;
+    register unsigned long a0 asm("a0") = arg1;
+    register unsigned long a1 asm("a1") = arg2;
 
     asm volatile(
        "movz $0, $0, $0"
-        : "+r"(a0)  // Input and output in R0
-        : "r"(a1) , "r" (a2)// arg1 in register A1
+        : "+r"(v0)  // Input and output in R0
+        : "r"(a0) , "r" (a1)// arg1 in register A1
```
Comment (be32826) [inline hypercall.h:76]: Fix (or delete) these comments

**Review by be32826 — DISMISSED** [review]
I had to fix MIPS issues with libhc recently, so I looked at the MIPS changes and they seem right.

**Review by AndrewFasano — APPROVED** [review]
Changes look good. A few questions:
* Is there an authoritative doc describing the ABI we're using here? If not can we make one (in libhc or somewhere else?) 
* Could these hypercall functions be made more generic (i.e., drop the `igloo_` prefix) and added to libhc? Then could we just include libhc in this repo? And then also in our kernel fork?

Just thinking about deduplicating code and ensuring we know what the standard we're using is

**Comment by caden-kline** [comment]
So this uses syscall register order(according the pypanda.arch.*). Yes I plan to make this generic but this is what's required to merge my penguin changes.  Generic version is the next step but I would like to close the hypercaller plugin issue first.

**Comment by AndrewFasano** [comment]
Looks like this fails to build in CI - I guess we could use better tests on PR to detect this sort of thing next time, *before* we merge it.

@caden-kline can you track this down? https://github.com/rehosting/libnvram/actions/runs/9962603228/job/27526759253
```
arm-linux-musleabi-gcc -c -DCONFIG_ARM=1 -O2 -fPIC -Wall -Wno-attribute-alias nvram.c -o nvram.o
In file included from nvram.c:825:
nvram.c: In function ‘strstr’:
strings.c:119:9: warning: ‘nonnull’ argument ‘haystack’ compared to NULL [-Wnonnull-compare]
  119 |     if (!haystack || !needle) return NULL;
      |         ^~~~~~~~~
strings.c:119:22: warning: ‘nonnull’ argument ‘needle’ compared to NULL [-Wnonnull-compare]
  119 |     if (!haystack || !needle) return NULL;
      |                      ^~~~~~~
arm-linux-musleabi-gcc -shared -nostdlib nvram.o -o libnvram.so
make: Leaving directory '/app'
+ mv nvram.o /tmp/tmp.9phlBDW7sP/libnvram/nvram.o.armel
+ mv libnvram.so /tmp/tmp.9phlBDW7sP/libnvram/libnvram.so.armel
+ make clean
rm -f *.o libnvram.so libnvram_ioctl.so
+ CC=arm-linux-musleabihf-gcc
+ make 'CFLAGS=-DCONFIG_ARM=1 -mfloat-abi=hard' libnvram.so -C /app
make: Entering directory '/app'
arm-linux-musleabihf-gcc -c -DCONFIG_ARM=1 -mfloat-abi=hard -O2 -fPIC -Wall -Wno-attribute-alias nvram.c -o nvram.o
In file included from nvram.c:[82](https://github.com/rehosting/libnvram/actions/runs/9962603228/job/27526759253#step:4:83)5:
nvram.c: In function ‘strstr’:
strings.c:119:9: warning: ‘nonnull’ argument ‘haystack’ compared to NULL [-Wnonnull-compare]
  119 |     if (!haystack || !needle) return NULL;
      |         ^~~~~~~~~
strings.c:119:22: warning: ‘nonnull’ argument ‘needle’ compared to NULL [-Wnonnull-compare]
  119 |     if (!haystack || !needle) return NULL;
      |                      ^~~~~~~
arm-linux-musleabihf-gcc -shared -nostdlib nvram.o -o libnvram.so
make: Leaving directory '/app'
+ mv nvram.o /tmp/tmp.9phlBDW7sP/libnvram/nvram.o.armelhf
+ mv libnvram.so /tmp/tmp.9phlBDW7sP/libnvram/libnvram.so.armelhf
+ make clean
rm -f *.o libnvram.so libnvram_ioctl.so
+ CC=aarch64-linux-musl-gcc
+ make CFLAGS=-DCONFIG_AARCH64=1 libnvram.so -C /app
make: Entering directory '/app'
aarch64-linux-musl-gcc -c -DCONFIG_AARCH64=1 -O2 -fPIC -Wall -Wno-attribute-alias nvram.c -o nvram.o
In file included from strings.h:4,
                 from nvram.c:23:
hypercall.h: In function ‘igloo_hypercall’:
hypercall.h:22:15: error: ‘x8’ undeclared (first use in this function); did you mean ‘x1’?
   22 |         : "r"(x8), "r"(x0)
      |               ^~
      |               x1
hypercall.h:22:15: note: each undeclared identifier is reported only once for each function it appears in
hypercall.h:15:33: warning: unused variable ‘x1’ [-Wunused-variable]
   15 |     register unsigned long long x1 asm("x0") = arg1;
      |                                 ^~
nvram.c: In function ‘nvram_get_buf’:
nvram.c:28:63: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
   28 | #define PRINT_MSG(fmt, ...) do { if (DEBUG) { fprintf(stderr, "%s: "fmt, __FUNCTION__, __VA_ARGS__); } } while (0)
      |                                                               ^~~~~~
nvram.c:306:9: note: in expansion of macro ‘PRINT_MSG’
  306 |         PRINT_MSG("\n\n[NVRAM] %d %s\n\n", strlen(key), key);
      |         ^~~~~~~~~
make: *** [Makefile:16: nvram.o] Error 1
make: Leaving directory '/app'
```

**Comment by caden-kline** [comment]
This is what I get for not building again after git changes. It looks like variable names got changed back when I did a rebase/merge and I didn't catch that.

**Pull request merged**